sensor: fxos8700: implement magnetic vector magnitude function

mtudan opened the pull request:
The magnetometer vector-magnitude function will generate an interrupt
when magnitude is greater than specified threshold value. The user
may program the threshold value to establish the conditions needed
to detect a magnetic vector-magnitude change event. Depending on the
values chosen for the reference values, this function may be
configured to detect a magnetic field magnitude that is above
a preset threshold (with reference values = 0), or a change in magnitude
between two magnetic vectors greater than the preset threshold
(with reference values non-zero).

Default configuration of M_VECM_CFG register (address 0x69) is 0x4e.

Signed-off-by: Matija Tudan <mtudan@mobilisis.hr>

MaureenHelm commented on drivers/sensor/fxos8700/fxos8700.h:
```diff
@@ -161,6 +176,7 @@ struct fxos8700_data {
 #ifdef CONFIG_FXOS8700_MOTION
 	sensor_trigger_handler_t motion_handler;
 #endif
+	sensor_trigger_handler_t m_vecm_handler;
```
Wrap this with `#ifdef CONFIG_FXOS8700_MAG_VECM..#endif`

MaureenHelm commented on drivers/sensor/fxos8700/fxos8700_trigger.c:
```diff
@@ -150,6 +169,27 @@ static void fxos8700_handle_int(void *arg)
+		fxos8700_handle_m_vecm_int(dev);
+	}
+
+	gpio_pin_interrupt_configure(data->gpio, config->gpio_pin,
```
I'm not sure it's right to do this again

MaureenHelm reviewed (changes requested):
Looks good. Just a nit and a question. Thanks for the contribution!

MaureenHelm reviewed (changes requested):
Please squash all the changes into one commit

Commit:
sensor: fxos8700: implement magnetic vector magnitude function

The magnetometer vector-magnitude function will generate an interrupt
when magnitude is greater than specified threshold value. The user
may program the threshold value to establish the conditions needed
to detect a magnetic vector-magnitude change event. Depending on the
values chosen for the reference values, this function may be
configured to detect a magnetic field magnitude that is above
a preset threshold (with reference values = 0), or a change in magnitude
between two magnetic vectors greater than the preset threshold
(with reference values non-zero).

Default configuration of M_VECM_CFG register (address 0x69) is 0x4e.

Signed-off-by: Matija Tudan <mtudan@mobilisis.hr>

mtudan commented:
Reorganized all changes into one commit.

gmarull commented on drivers/sensor/fxos8700/Kconfig:
```diff
@@ -209,6 +209,42 @@ config FXOS8700_MOTION_INT1
+	default 0x5a
+	help
+	  Eight least significant bits of 15-bit unsigned magnetic
+	  vector-magnitude threshold. Resolution is 0.1 Î¼T/LSB.
```
I think that initial hardware configuration should use device tree bindings. Using device tree bindings allows to have per-device settings. Maybe this is not needed for this type of sensor, but who knows on specific applications.

mtudan replied:
Should we then wait for PR #27418 to be merged and realign with that or what do you suggest?

MaureenHelm replied:
No, I don't think that PR will be ready for a bit. I propose we go ahead with this in Kconfig for now, and I can roll in the vector magnitude bits into the larger Kconfig-to-devicetree conversion. @gmarull are you ok with that?

gmarull replied:
I'm ok with that, no problem

MaureenHelm replied:
Thanks. Mind giving your +1 on this PR?

gmarull reviewed (commented):
LGTM, just left a comment regarding Kconfig vs DT
